meta-balena-common/recipes-connectivity/modemmanager/balena-files/u-blox-switch.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+set -e
+MODEM_TTY_PATH=$1
+
+while true; do
+        echo -e "AT+UUSBCONF=2,\"ECM\"\r\n" > $MODEM_TTY_PATH
+        sleep 2
+        echo -e "AT+CFUN=1,1\r\n" > $MODEM_TTY_PATH
+        sleep 20
+        if lsusb | grep "1546:1143" > /dev/null; then # test if modem is in ECM mode
+                echo "Succesfully switched modem mode"
+                reboot
```

Comment:
Even though this reboot comes from the balena-raspberry repo, I'd rather not do this. The modem firmware is likely to reset itself after the mode change, and this mode change should be happening before MM tries to use the modem, so let's test whether this works without the reboot.

Reply:
I have established that the `reboot` after a mode change isn't necessary. However, testing has shown that the`u-blox-switch.sh` script works less than 25% of the time. When it fails the system continuously tries to change the modem mode which prevents any networking from starting.

Testing was conducted using a balena Fin with a u-blox Toby L200 modem. A test script was written to revert the modem to RNDIS mode. Once in RNDIS mode the system was rebooted and monitored via a serial link to determine whether `u-blox-switch.sh` had run correctly. Minicom was also deployed to the target to allow direct interaction with the modem device for debugging purposes.

If this PR is used to move u-blox-switch.sh from balena-raspberrypi to meta-balena as originally planned, I will open a separate issue/PR to address to problems with the script itself.

Reply:
@markcorbinuk could you please cross link the new issue to this PR please?

Reply:
@alexgg cross-linked to #2039